core/src/main/java/org/elasticsearch/gateway/PrimaryShardAllocator.java

```diff
@@ -161,11 +243,11 @@ private boolean isEnoughAllocationsFound(ShardRouting shard, IndexMetaData index
     /**
      * Based on the nodes and versions, build the list of yes/no/throttle nodes that the shard applies to.
```

Comment:
comment should be - Split the list of nodes to lists of yes/no/throttle nodes based on allocation deciders (versions are now irrelevant)